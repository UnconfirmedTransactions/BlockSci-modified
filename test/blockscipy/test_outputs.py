import pytest


def test_p2pkh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2pkh-spend-2-tx"])
    output_regression(tx, regtest)


def test_p2sh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2sh-spend-0-tx"])
    output_regression(tx, regtest)


@pytest.mark.btc
@pytest.mark.ltc
def test_p2wpkh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2wpkh-spend-1-tx"])
    output_regression(tx, regtest)


@pytest.mark.btc
@pytest.mark.ltc
def test_p2wsh_output_regression(chain, json_data, regtest):
    tx = chain.tx_with_hash(json_data["address-p2wsh-spend-1-tx"])
    output_regression(tx, regtest)


def output_regression(tx, regtest):
    for out in tx.outputs:
        print(out.address, file=regtest)
        print(out.address_type, file=regtest)
        print(out.block, file=regtest)
        print(out.index, file=regtest)
        print(out.is_spent, file=regtest)
        print(out.spending_input, file=regtest)
        print(out.spending_tx, file=regtest)
        print(out.spending_tx_index, file=regtest)
        print(out.tx, file=regtest)
        print(out.tx_index, file=regtest)
        print(out.value, file=regtest)

def test_output_input_mapping(chain):
    for block in chain:
        for tx in block:
            for output in tx.outputs:
                if output.is_spent:
                    assert output == output.spending_input.spent_output
                    assert tx == output.spending_input.spent_tx
