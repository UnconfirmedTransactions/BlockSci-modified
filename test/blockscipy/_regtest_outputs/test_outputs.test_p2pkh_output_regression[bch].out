PubkeyHashAddress(mk7GYZDvnyyTyi2anTr5w5sTqW3r18WQWb)
Pay to pubkey hash
Block(tx_count=5, height=113, header_hash=121d93886d8ec313c13465d4cb9421fbf63deea028c0e14315a4120c74750f19, version=536870912, timestamp=1535762400, bits=545259519, nonce=4)
0
True
TxIn(spent_tx_index=118, address=PubkeyHashAddress(mk7GYZDvnyyTyi2anTr5w5sTqW3r18WQWb), value=100000000)
Tx(len(txins)=1, len(txouts)=1, size_bytes=191, block_height=113, tx_index=119)
119
Tx(len(txins)=1, len(txouts)=2, size_bytes=226, block_height=113, tx_index=118)
118
100000000
PubkeyHashAddress(mjTkW3DjgyZck4KbiRusZsqTgaYTxdSz6z)
Pay to pubkey hash
Block(tx_count=5, height=113, header_hash=121d93886d8ec313c13465d4cb9421fbf63deea028c0e14315a4120c74750f19, version=536870912, timestamp=1535762400, bits=545259519, nonce=4)
1
False
None
None
None
Tx(len(txins)=1, len(txouts)=2, size_bytes=226, block_height=113, tx_index=118)
118
4899990000
